bin/generate_builtins.jl

```diff
@@ -246,6 +250,24 @@ function maybe_evaluate_builtin(frame, call_expr, expand::Bool)
             return Some{Any}(Core.eval(moduleof(frame), call_expr))
         end
 """)
+    # recently removed builtins
+    for (; mod, name) in RECENTLY_REMOVED
+        minarg = 1
+        if name in (:arrayref, :const_arrayref)
+            maxarg = 5
+        elseif name === :arrayset
+            maxarg = 6
+        elseif name === :arraysize
+            maxarg = 2
+        end
+        fcall = generate_fcall_nargs(name, minarg, maxarg)
+        name = repr(name)
+        print(io,
+"""
+    elseif @static isdefined($mod, $name) && f === $name
```

Comment:
```suggestion
        rname = repr(name)
        print(io,
"""
    elseif @static isdefined($mod, $rname) && f === $name
```
? Otherwise `f === :arrayref` does never seem to be true?

Reply:
Also, they are now generic functions defined within `Core`, so e.g. `isdefined(Core, :arrayref)` will be `true` on the latest master. Don't we want to have `Core.arrayref isa Core.Builtin` instead?

Reply:
> Also, they are now generic functions defined within Core, so e.g. isdefined(Core, :arrayref) will be true on the latest master. Don't we want to have Core.arrayref isa Core.Builtin instead?

My thought process was that even though they're not technically builtins anymore there isn't really any point in stepping into them and we might as well evaluate them eagerly

Reply:
That sounds fine. Let's move this forward as is.